Confluence v2 implementation (Cloud)

batzel opened the pull request:
This PR implements full support for Confluence Cloud v2 API including:

- Core API structure and endpoint mapping
- Page operations (get, create, update, delete)
- Space operations
- Search functionality
- Comments handling
- Whiteboard and custom content types (new in v2)
- Comprehensive compatibility layer for v1 to v2 migration
- Documentation and examples
- Testing framework

Note backward compatibility for api v1. Also tested by hand with mcp-atlassian.
Implementation by John B Batzel (batzel@upenn.edu) with assistance from Cursor.com AI (or vice-versa).

gonchik commented:
Looks like confluence module put into dedicated directory like bitbucket, otherwise too many files will be :)

batzel commented:
I fixed the earlier automatic security reports, moved confluence to a separate directory like you wanted, and I grabbed the two commits you made to master and incorporated them into the new confluence directory structure.

batzel commented:
Superceeding this one with a different pull request.
